projects/distributed-tracing/src/shared/dashboard/interaction/detail-sheet/detail-sheet-interaction-handler.model.ts

```diff
@@ -0,0 +1,48 @@
+import { SheetSize } from '@hypertrace/components';
+import { EnumPropertyTypeInstance, ENUM_TYPE, ModelTemplatePropertyType } from '@hypertrace/dashboards';
+import { Model, ModelApi, ModelJson, ModelProperty } from '@hypertrace/hyperdash';
+import { ModelInject, MODEL_API } from '@hypertrace/hyperdash-angular';
+import { Observable, of } from 'rxjs';
+import { InteractionHandler } from '../interaction-handler';
+import { DetailSheetInteractionHandlerService } from './detail-sheet-interaction-handler.service';
+
+@Model({
+  type: 'detail-sheet-interaction-handler'
+})
+export class DetailSheetInteractionHandlerModel implements InteractionHandler {
+  @ModelProperty({
+    key: 'detail-template',
```

Comment:
naming comment : 
can we use a more generic name here?
This interaction handler could be used for non detail views as well. 

Perhaps just `OverlaySheetInteractionHandlerModel`, `sheet-template` etc?

Reply:
So, I kept the name detail because of its ability to provide the original data which was interacted with(row in case of table). So just with just overlay sheet it is not obv. I am happy to rename this, but i want it to convey that this model injects a context data.